spec/workers/company_domain_worker_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe CompanyDomainWorker, sidekiq: true do
+  let!(:company) { create(:company, :worker_domain) }
+  let!(:worker) { CompanyDomainWorker.new }
+
+  before(:each) do
+    allow(worker).to receive(:link_open).and_return(response_html)
+
+    worker.perform(company.id)
+  end
+
+  describe 'with valid data' do
+    let!(:response_html) { Nokogiri::HTML(open('spec/factories/test.html')) }
+
+    it 'testing worker queueing' do
+      expect { CompanyDomainWorker.perform_async(1) }.to change(CompanyDomainWorker.jobs, :size).by(1)
+    end
+
+    it 'worker company should be as created one' do
+      expect(worker.company).to eq(company)
+    end
+
+    it 'sub_links content should be correct. not nil, empty, outside' do
+      worker.send(:filtered_sub_pages_links).each do |link|
+        expect(link).not_to be_nil
+        expect(link).not_to be_empty
+        expect(link).not_to be('#')
+
+        expect(link).to include(worker.domain)
+      end
+    end
+
+    it 'number of created pages should be as number of sub_links' do
+      expect { worker.perform(company.id) }.to change(Page, :count).by(89)
+    end
+  end
+
+  describe 'with invalid data' do
+    let!(:response_html) do
+      Nokogiri::HTML('<html><head><title>Hello World</title></head>
+                      <body><a href="#"></a>
+                            <a href=""></a>
+                            <a href=></a>
+                            <a href="http://youtube.com"></a></body></html>')
+    end
+
+    it 'empty sub_links array' do
+      expect(worker.send(:filtered_sub_pages_links)).to be_empty
```

Comment:
this looks really strange.. usually if you test private or protected method it means that something not right..

Reply:
wanted to check if the array of links we get from html page, to make sure that there is no invalid data in it. to see that filtering goes right. previously had an attr_reader for the array but there is no point for it only for testing..

Reply:
Add in this spec links to external sites. So expected array will be empty too

Reply:
testing data for this spec has youtube.com as external link

`let!(:response_html) do
      Nokogiri::HTML('<html><head><title>Hello World</title></head>
                      <body><a href="#"></a>
                            <a href=""></a>
                            <a href=></a>
                            <a href="http://youtube.com"></a></body></html>')
    end`